### ISSUE TITLE
Delete without trash

### ISSUE THREAD

**Issue by notramo** [jurplel/qView#497]
On Linux with BTRFS, NILFS2, and ZFS, it's not necessary to have a thrash, since these filesystems support snapshots.
On CLI systems which `rm` without thrash, it's unnecessary complication to have a thrash just for qView, and clean it periodically.
qView should have an option to disable trash (enable direct deletion) in the settings.

**Comment by A9G-Data-Droid** [comment]
I am unable to delete files from an SMB share with no trash. This might be the reason. Instead of throwing an error, we need to issue a regular filesystem delete on SMB shares.

**Comment by vassudanagunta** [comment]
> I am unable to delete files from an SMB share with no trash.

I am also unable to delete files from an SMB share, but it *does* have a trash, at least on the server side (It's just another Mac).

Anyway, if the inability to move files to the trash on SMB shares is what's preventing deletion, then I agree with @A9G-Data-Droid, as long as there a special confirmation dialogue. I can have a checkbox for "don't ask me again".

**Comment by jurplel** [comment]
I can't believe this isn't in qView currently. If someone wants to implement this, just make a new action called "force delete" or something, absent from the context menu but present in the menubar, with a default keyboard shortcut of cmd/ctrl+del and for macOS cmd+option+backspace, or something along those lines. maybe you can add a modifier key to the `QKeySequence::delete` or something also.

**Comment by simonbcn** [comment]
I fully support this improvement. So deleting files on network drives would not be a problem.

**Comment by vassudanagunta** [comment]
We have a pull request that fixes this, #639, that's 18 months old. @jurplel not complaining, as i know you're busy and doing this for free. but i do think it impacts a lot of people, as new issues about this keep getting created.

**Comment by jurplel** [comment]
> We have a pull request that fixes this, #639, that's 18 months old. @jurplel not complaining, as i know you're busy and doing this for free. but i do think it impacts a lot of people, as new issues about this keep getting created.

Yes, I absolutely need to merge it. I'm going to try to update qView again in a couple of weeks, but I am trying to coordinate a move during finals week right now.

**Comment by vassudanagunta** [comment]
good luck on your finals!

**Comment by jurplel** [comment]
Gonna try to make a release soon, yes.

**Comment by peenstra** [comment]
Is there any progress on this issue? Or updates? Thanks!

**Comment by jurplel** [comment]
Sorry, still need to do this. I'll try to set some time aside for break in a couple weeks

**Comment by peenstra** [comment]
> Sorry, still need to do this. I'll try to set some time aside for break in a couple weeks

Did you have any change to do it =)

**Comment by 4N7** [comment]
Same issue for me -- using an NFS mount.
The file can be deleted using finder, after the "Are you sure?" prompt is answered.

![Image](https://github.com/user-attachments/assets/19ceb603-8cc6-4b9b-aed2-2d8d168a352a)

### Build
```
$ git describe --tags --long
nightly-7-g96105cc

$ git rev-parse HEAD
96105cc87a699fe847eb9876d92303babcb968da
```

![Image](https://github.com/user-attachments/assets/af27a17b-7f68-44c4-9a58-b4c69a731ee8)

**Comment by ghost** [comment]
Why this issue is closed????? The basic file operation like delete should be worked in day 1.

- https://github.com/jurplel/qView/issues/832
- https://github.com/jurplel/qView/issues/840
- https://github.com/jurplel/qView/issues/832

**Comment by vassudanagunta** [comment]
@frugalion,

> The basic file operation like delete should be worked in day 1.

Software is complex so don't assume anything is easy, especially for software that must work on multiple platforms.

But **far more importantly**, this is open source and free. The developer is doing this out of the kindness of his heart. Don't be rude and demanding. You are entitled to nothing. If you think it is so easy, why don't you open a pull request to fix whatever you think is broken? From the looks of your account, you have contributed nothing, and created the account only to complain and make demands on people.

**Comment by jurplel** [comment]
Sorry, but I think this feature is implemented in qview. There's a delete permanently keybind available.

This looks like an issue with nfs mounts? Which would be a separate issue. But I haven't tested this, it deserves some attention.

Apologies for not commenting on all of these issues, but the volume is such that it's hard to keep up in the limited time I have outside of work and other commitments.
